packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,266 @@
+                    {this.renderItems()}
+                </Tabs>
+                <Flipper
+                    direction={FlipperDirection.next}
+                    onClick={this.handleNextInvoke}
+                    className={get(this.props, "managedClasses.carousel_flipperNext", "")}
+                />
+            </div>
+        );
+    }
+
+    /**
+     * Generate class names
+     */
+    protected generateClassNames(): string {
+        let className: string = get(this.props, "managedClasses.carousel", "");
+        let theme: string = "";
+
+        if (this.getSlideTheme()) {
+            theme =
+                this.getSlideTheme() === SlideTheme.light
+                    ? get(this.props, "managedClasses.theme__light", "")
+                    : get(this.props, "managedClasses.theme__dark", "");
+
+            className += ` ${theme}`;
+        }
+
+        return super.generateClassNames(className);
+    }
+
+    /**
+     * Get the active slide index
+     */
+    private getActiveIndex(): number {
+        return this.slides.map((slide: Slide) => slide.id).indexOf(this.state.activeId);
+    }
+
+    /**
+     * Get the current slide theme (needed for UI overlay contrast)
+     */
+    private getSlideTheme(): string {
+        if (!this.slides || this.slides.length <= 0) {
+            return;
+        }
+
+        return this.slides[this.getActiveIndex()].theme
+            ? this.slides[this.getActiveIndex()].theme
+            : SlideTheme.dark;
+    }
+
+    /**
+     * Updates slides
+     */
+    private updateSlides(): void {
+        if (this.props.items.length > 0) {
+            this.slides = this.props.items.map((item: Slide): Slide => item);
+        }
+    }
+
+    /**
+     * Change active tab
+     */
+    private handleUpdate = (activeTab: string): void => {
+        this.setState({
+            activeId: activeTab,
+        });
+    };
+
+    /**
+     * Move to previous slide if applicable
+     */
+    private handlePreviousInvoke = (): void => {
+        let newPosition: number = this.getActiveIndex() - 1;
+
+        if (newPosition < 0) {
+            newPosition = this.slides.length - 1;
+        }
+
+        this.setNewSlidePosition(newPosition);
+    };
+
+    /**
+     * Move to next slide if applicable
+     */
+    private handleNextInvoke = (): void => {
+        let newPosition: number = this.getActiveIndex() + 1;
+
+        if (newPosition > this.slides.length - 1) {
+            newPosition = 0;
+        }
+
+        this.setNewSlidePosition(newPosition);
+    };
+
+    /**
+     * Sets new slide based on position
+     */
+    private setNewSlidePosition(position: number): void {
+        this.setState({
+            activeId: this.slides[position].id,
+        });
+    }
+
+    /**
+     * Renders the tab item
+     */
+    private renderItems = (): JSX.Element[] => {
+        if (!this.slides || this.slides.length <= 0) {
+            return;
+        }
+
+        return this.slides.map(
```

Comment:
I could be wrong here but i don't think you need to map and return a `<TabItem/>` we are already doing that in the tabs base component. Really all you need to do is style it to layout the way you want. So you could just map the classNames to your carousel classNames like:

```
private tabsManagedClasses(): TabsClassNameContract {
        return {
            tabs: this.props.managedClasses.carousel_tabs,
            tabs_tabPanels: this.props.managedClasses.carousel_tabs_tabPanels,
            tabs_tabList: this.props.managedClasses.carousel_tabs_tabList,
        };
    }
```

Reply:
Yep yep. After the fix went in yesterday to Tabs, this is working right here.